packages/shared/lib/core/wallet/utils/generateActivity/generateActivitiesFromAccountOutputs.ts

```diff
@@ -0,0 +1,25 @@
+import { IWalletState } from '@core/wallet/interfaces'
+import { ActivityAction, EMPTY_HEX_ID, IProcessedTransaction } from '@core/wallet'
+import { Activity, type AccountActivity } from '@core/wallet/types'
+import { generateSingleAccountActivity } from './generateSingleAccountActivity'
+import { AccountOutput, OutputType } from '@iota/sdk/out/types'
+
+export async function generateActivitiesFromAccountOutputs(
+    processedTransaction: IProcessedTransaction,
+    wallet: IWalletState
+): Promise<Activity[]> {
+    const outputs = processedTransaction.outputs
+    const activities: AccountActivity[] = []
+
+    const accountOutputs = outputs.filter((output) => output.output.type === OutputType.Account)
+    for (const accountOutput of accountOutputs) {
+        const output = accountOutput.output as AccountOutput
+        const activity = await generateSingleAccountActivity(wallet, {
+            action: output.accountId === EMPTY_HEX_ID ? ActivityAction.Mint : ActivityAction.Send,
```

Comment:
Where the condition should be updated is here.
imo, for now, I would put this
`action: output.accountId === EMPTY_HEX_ID ? ActivityAction.Send : ActivityAction.Mint,`

I would add a comment and update the issue to check this logic